# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gcc/no-late-combine-for-ia64.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- gcc-15.2.0/gcc/config/ia64/ia64.cc.vanilla	2025-12-05 20:42:45.202315304 +0100
+++ gcc-15.2.0/gcc/config/ia64/ia64.cc	2025-12-05 20:43:02.960046640 +0100
@@ -6153,6 +6153,10 @@
          a transformation.  */
       flag_auto_inc_dec = 0;
     }
+
+  /* Disable late-combine by default, it produces broken code */
+  if (!OPTION_SET_P (flag_late_combine_instructions))
+    flag_late_combine_instructions = 0;
 }
 
 /* Initialize the record of emitted frame related registers.  */
